### PR TITLE
ROU-4536: Improve visualViewport validation

### PR DESCRIPTION
**Pull request by BenOsodrac**
This PR is for improving the visualViewport validation, to prevent errors when its undefined.

### Checklist

-   [x] tested locally
-   [x] documented the code
-   [x] clean all warnings and errors of eslint
-   [x] requires changes in OutSystems (if so, provide a module with changes)
-   [ ] requires new sample page in OutSystems (if so, provide a module with changes)

### CODE REVIEW

**Review by bmarcelino-fe — APPROVED** [review]

File: src/scripts/OSFramework/Helper/Device.ts [hunk 1]
```diff
@@ -347,8 +347,8 @@ namespace OSFramework.Helper {
 						? sessionStorage.previewDevicesPixelRatio
 						: window.devicePixelRatio) || 1;
 				const currScreen: iphoneDetails = {
-					width: window.visualViewport.width * ratio,
-					height: window.visualViewport.height * ratio,
+					width: window.visualViewport ? window.visualViewport.width : window.innerWidth * ratio,
```
Comment (gnbm) [inline src/scripts/OSFramework/Helper/Device.ts:350]: @BenOsodrac Wouldn't it be easier to read if we had some parenthesis, like:
`(window.visualViewport ? window.visualViewport.width : window.innerWidth) * ratio`

Comment (BenOsodrac) [reply src/scripts/OSFramework/Helper/Device.ts]: Made the change!

**Comment by gnbm** [comment]
@BenOsodrac Why was this PR closed, since the task is still in code review?

**Comment by BenOsodrac** [comment]
@gnbm because @rugoncalves created a new odc main, and due to hundreds of conflicts, was easier to create a new branch and PR, than try the rebase and merge

**Pull request closed without merging**